File: investsys_front/src/services/customerService.js
import axios from 'axios';

const API_URL = 'http://localhost:8080/investsys/customer';

const getAllCustomers = () => {
    return axios.get(API_URL);
};

const addCustomer = (customer) => {
    return axios.post(API_URL, customer);
};

const getCustomer = (customerID) => {
    return axios.get(`${API_URL}/${customerID}`);
}

const deleteCustomer = (customerID) => {
    return axios.delete(`${API_URL}/${customerID}`);
}

export default {
    getAllCustomers,
    addCustomer,
    getCustomer,
    deleteCustomer
};

File: investsys_back/src/main/java/com/citi/investSys/service/CustomerService.java
package com.citi.investSys.service;

import com.citi.investSys.model.Customer;

import java.util.List;

public interface CustomerService {
    String addCustomer(Customer customer);

    String removeCustomer(String customerID);

    Customer getCustomer(String customerID);

    List<Customer> getAllCustomers();
}
